src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/validators/VisitorCountValidator.kt

```diff
@@ -0,0 +1,37 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.controller.validators
+
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitorDto
+import javax.validation.Constraint
+import javax.validation.ConstraintValidator
+import javax.validation.ConstraintValidatorContext
+import javax.validation.Payload
+import kotlin.reflect.KClass
+
+@Target(AnnotationTarget.FIELD)
+@MustBeDocumented
+@Constraint(validatedBy = [VisitorCountValidator::class])
+annotation class VisitorCountValidation(
+  val message: String = "{javax.validation.constraints.visitor.count.message}",
+  val groups: Array<KClass<*>> = [],
+  val payload: Array<KClass<out Payload>> = [],
+  val maximumVisitorsAllowed: UShort = VisitorCountValidator.MAXIMUM_VISITORS_ALLOWED,
+)
+
+class VisitorCountValidator : ConstraintValidator<VisitorCountValidation, Set<VisitorDto>> {
+  companion object {
+    const val MAXIMUM_VISITORS_ALLOWED: UShort = 10u
+  }
+  private var maximumVisitorsAllowed: UShort = MAXIMUM_VISITORS_ALLOWED
+
+  override fun initialize(constraintAnnotation: VisitorCountValidation?) {
+    constraintAnnotation?.let {
+      this.maximumVisitorsAllowed = constraintAnnotation.maximumVisitorsAllowed
+    }
+  }
+
+  override fun isValid(childList: Set<VisitorDto>?, context: ConstraintValidatorContext?): Boolean {
+    // Cannot have more than max visitors allowed
+    val total = (childList?.size ?: 0).toUShort()
+    return (total <= maximumVisitorsAllowed)
+  }
+}
```

Comment:
I would do the following
```

package uk.gov.justice.digital.hmpps.visitscheduler.controller.validators

import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitorDto
import javax.validation.Constraint
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext
import javax.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD)
@MustBeDocumented
@Constraint(validatedBy = [VisitorCountValidator::class])
annotation class VisitorCountValidation(
  val message: String = "{javax.validation.constraints.visitor.count.message}",
  val groups: Array<KClass<*>> = [],
  val payload: Array<KClass<out Payload>> = [],
  val maximumVisitorsAllowed: Int = 10,
)

class VisitorCountValidator : ConstraintValidator<VisitorCountValidation, Set<VisitorDto>> {

  private var maximumVisitorsAllowed: Int = 0

  override fun initialize(constraintAnnotation: VisitorCountValidation) {
      this.maximumVisitorsAllowed = constraintAnnotation.maximumVisitorsAllowed
  }

  override fun isValid(visitors: Set<VisitorDto>?, context: ConstraintValidatorContext?): Boolean {
    // Cannot have more than max visitors allowed
    val totalVisitors = visitors?.size ?: 0
    return (totalVisitors <= maximumVisitorsAllowed)
  }
}


```

Just to make it simple and straight forward

Reply:
Thanks, updated based on your comments.